java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/MemoryRecordTests.java

```diff
@@ -0,0 +1,270 @@
+        assertEquals(_embedding.getVector(), memoryRecord.getEmbedding().getVector());
+    }
+
+    @Test @Disabled("JSON deserialization is not implemented")
+    void itCanBeDeserializedFromJson()
+    {
+        // Arrange
+        String jsonString = "{"
+                + "  \"metadata\" : {"
+                + "    \"is_reference\" : false"
+                + "    \"id\" : \"Id\""
+                + "    \"text\" : \"text\""
+                + "    \"description\" : \"description\""
+                + "    \"external_source_name\" : \"externalSourceName\""
+                + "    \"additional_metadata\" : \"value\""
+                + "  }"
+                + "  \"embedding\" : {"
+                + "    \"vector\" : [ 1, 2, 3 ]"
+                + "  }"
+                + "}";
+
+        // Act
+        MemoryRecord memoryRecord = null; //JsonSerializer.Deserialize<MemoryRecord>(jsonString);
+
+        // Assert
+        assertNotNull(memoryRecord);
+        assertEquals(_isReference, memoryRecord.getMetadata().isReference());
+        assertEquals(_id, memoryRecord.getMetadata().getId());
+        assertEquals(_text, memoryRecord.getMetadata().getText());
+        assertEquals(_description, memoryRecord.getMetadata().getDescription());
+        assertEquals(_externalSourceName, memoryRecord.getMetadata().getExternalSourceName());
+        assertEquals(_externalSourceName, memoryRecord.getMetadata().getExternalSourceName());
+        assertEquals(_embedding.getVector(), memoryRecord.getEmbedding().getVector());
+    }
+
+    @Test @Disabled("JSON serialization is not implemented")
+    void itCanBeSerialized()
+    {
+        // Arrange
+        String jsonString = "{"
+                + "  \"metadata\" : {"
+                + "    \"is_reference\" : false"
+                + "    \"id\" : \"Id\""
+                + "    \"text\" : \"text\""
+                + "    \"description\" : \"description\""
+                + "    \"external_source_name\" : \"externalSourceName\""
+                + "    \"additional_metadata\" : \"value\""
+                + "  }"
+                + "  \"embedding\" : {"
+                + "    \"vector\" : [ 1, 2, 3 ]"
+                + "  }"
+                + "\"key\" : \"key\""
+                + "\"timestamp\": null"
+                + "}";
+
+        MemoryRecordMetadata metadata = new MemoryRecordMetadata(
+            _isReference,
+            _id,
+            _text,
+            _description,
+            _externalSourceName,
+            _additionalMetadata);
+        MemoryRecord memoryRecord = new MemoryRecord(metadata, _embedding, "key", ZonedDateTime.now());
+
+        // Act
+        String serializedRecord = null;//JsonSerializer.Serialize(memoryRecord);
+        jsonString = jsonString.replaceAll("\n", "");
+        jsonString = jsonString.replaceAll(" ", "");
+
+        // Assert
+        assertEquals(jsonString, serializedRecord);
```

Comment:
Think this should always fail, jsonString cant be null and serializedRecord must be null